lib/wallet/wallet.js

```diff
@@ -107,7 +108,12 @@ class Wallet extends EventEmitter {
       assert(HD.isPrivate(key),
         'Must create wallet with hd private key.');
     } else {
-      mnemonic = new Mnemonic(options.mnemonic);
+      if (!(mnemonic instanceof Mnemonic)) {
```

Comment:
so to reiterate, this could be rewritten to be more precise to our use case (to be more true to what gets passed around)
and I think it makes this more readable.
```js
  if (typeof mnemonic === 'string')
    mnemonic = new Mnemonic({ phrase: mnemonic });

  // null or undefined
  if (mnemonic == null) // or if (!mnemonic) to include all falsy values?
    mnemonic = new Mnemonic({ language: options.language }); 
```

Reply:
agreed, thanks - I cleaned this up